pkg/cmd/server/origin/controller/apps.go

```diff
@@ -0,0 +1,90 @@
+package controller
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	kapi "k8s.io/kubernetes/pkg/api"
+
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	deployercontroller "github.com/openshift/origin/pkg/deploy/controller/deployer"
+	deployconfigcontroller "github.com/openshift/origin/pkg/deploy/controller/deploymentconfig"
+	triggercontroller "github.com/openshift/origin/pkg/deploy/controller/generictrigger"
+)
+
+type DeployerControllerConfig struct {
+	ImageName     string
+	ClientEnvVars []kapi.EnvVar
+
+	Codec runtime.Codec
+}
+
+type DeploymentConfigControllerConfig struct {
+	Codec runtime.Codec
+}
+
+type DeploymentTriggerControllerConfig struct {
+	Codec runtime.Codec
+}
+
+func (c *DeployerControllerConfig) RunController(ctx ControllerContext) (bool, error) {
+	internalDeployerKubeClient, err := ctx.ClientBuilder.KubeInternalClient(bootstrappolicy.InfraDeployerControllerServiceAccountName)
+	if err != nil {
+		return false, err
```

Comment:
`return true, err` . You were enabled and you failed.  Returning false means you had some secondary bit of data that said "I don't need to start".  Discovery revealing that your resources were missing perhaps?  A check for  a later time I think.